.github/workflows/examples-imports-check.yml

```diff
@@ -0,0 +1,10 @@
+name: Check examples don't import internal modules
+
+on: push
+
+jobs:
+  examples-imports-check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - run: bash .github/workflows/scripts/examples-imports-check.sh
```

Comment:
I'd prefer this to be available in the makefile and run in hydra like `formatting-check` in `flake.nix`, rather than a github action